src/main/java/cart/web/config/PathVariableId.java

```diff
@@ -0,0 +1,36 @@
+package cart.web.config;
+
+public class PathVariableId {
+    private static final int MIN_PATH_VARIABLE_ID = 1;
+
+    private final Long id;
+
+    public PathVariableId(final Long id) {
+        validate(id);
+        this.id = id;
+    }
+
+    private void validate(final Long id) {
+        if (id < MIN_PATH_VARIABLE_ID) {
+            throw new IllegalArgumentException();
+        }
+    }
+
+    public static PathVariableId valueOf(final String value) {
+        final Long longValue = parseId(value);
+
+        return new PathVariableId(longValue);
+    }
+
+    private static Long parseId(final String value) {
+        try {
+            return Long.parseLong(value);
+        } catch (final NumberFormatException exception) {
+            throw new IllegalArgumentException("경로에 포함된 Id는 숫자만 가능합니다.");
```

Comment:
웹 환경으로 오면서 자바 내장 예외를 사용하는 것은 안좋게 느껴지는데 어떻게 생각하시나요?

1레벨때는 라이브러리가 적게 사용되기에, 예외가 터지는 부분을 쉽게 파악해서 직접 다뤄주기가 가능하지만, 프레임워크에서 발생하는 수많은 IllegalArgumentException 과, 우리의 IllegalArgumentException 을 분리해서 핸들링 하기 어렵다는 문제가 있어보여요

Reply:
추가로 exception 을 다른 예외로 전환하는 과정으로 보이는데요 이럴때는
```suggestion
            throw new IllegalArgumentException("경로에 포함된 Id는 숫자만 가능합니다.",exception);
```
형태로 바꿔보면 어떨까요?

Reply:
- `웹 환경으로 오면서 자바 내장 예외를 사용하는 것은 안좋게 느껴지는데 어떻게 생각하시나요?`
동의합니다. 위 예시처럼 `NumberFormatException`가 우리(코드 작성자..?)의 논리로 전환하는 과정 역시 단순히 메세지만 담을 뿐,`java.lang.exception`을 사용하는건 어플리케이션 내 다양한 예외 사항에 대해 디버깅하고 핸들링하기 어렵다는 문제가 있죠.

그렇다면, 커스텀 예외를 작성하여 다양한 예외 사항에 대해 세분화 하고 각각의 상황에 대한 직관적인 명시를 통해 위 문제를 개선할 수 있을 것 같습니다. 장단에 대해 이해 하였지만, 해당 사항을 반영하기에 여유...가 있다면 반영하겠습니다!

상황에 따른 커스템 예외 정의 - 메세지 작성 - 정의한 예외 사용 - handler에서 커스템 예외 처리 - 일관적인 응답 형식 구성 등의 작업이 필요할 것 같네요